Libplanet/Net/Swarm.BlockSync.cs

```diff
@@ -16,436 +20,556 @@ public partial class Swarm<T>
         /// It is null when the <see cref="Swarm{T}"/> does not have any block to demand.
         /// <seealso cref="BlockDemand"/>
         /// </summary>
-        public BlockDemand? BlockDemand { get; private set; }
+        public ConcurrentDictionary<BoundPeer, BlockDemand> BlockDemands { get; private set; }
```

Comment:
From outside, wouldn't it better be `IDictionary<BoundBeer, BlockDemand>` so that its concrete implementation is not exposed?

```suggestion
        public IDictionary<BoundPeer, BlockDemand> BlockDemands { get; private set; }
```

Reply:
I generally agree with the statement, but ultimately, I think it would be preferable to have its own internal logic as a class, be it `BlockDemands` and/or `BlockDemand`. 🙃

Reply:
I've added `BlockDemandTable` class. PTAL :)